src/readmeoss/agent/readmeoss.php

```diff
@@ -141,11 +141,38 @@ function processUploadId($uploadId)
     }
 
     $contents = array('licensesMain'=>$licenseStmtsMain, 'licenses'=>$licenseStmts, 'copyrights'=>$copyrightStmts, 'licenseAcknowledgements' => $licenseAcknowledgements);
+    $contents = $this->identifiedGlobalLicenses($contents);
     $this->writeReport($contents, $uploadId);
 
     return true;
   }
 
+  /**
+    * @brief Copy identified global licenses
+    * @param[int,out] array $contents
+    * @return array $contents with identified global license path
+    */
+  function identifiedGlobalLicenses($contents)
+  {
+    foreach ($contents["licensesMain"] as $key1 => $value1) {
+      $flag = 0;
```

Comment:
`$flag` is a very generic name, and should be replaced by a more descriptive one